nautobot/dcim/filters/__init__.py

```diff
@@ -804,6 +830,12 @@ class DeviceFilterSet(
         to_field_name="name",
         label="Manufacturer (name or ID)",
     )
+    hardware_family = NaturalKeyOrPKMultipleChoiceFilter(
+        field_name="device_type__hardware_family",
```

Comment:
Similarly probably want to add a select_related for this lookup on DeviceListView as well.